dataactbroker/scripts/update_historical_duns.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import boto
+import os
+import pandas as pd
+import argparse
+from datetime import datetime
+
+from dataactcore.models.domainModels import DUNS
+from dataactcore.scripts.loadDUNS import load_duns_by_row
+from dataactvalidator.scripts.loaderUtils import clean_data
+from dataactvalidator.health_check import create_app
+from dataactcore.interfaces.db import GlobalDB
+from dataactcore.logging import configure_logging
+from dataactcore.config import CONFIG_BROKER
+
+logger = logging.getLogger(__name__)
+
+# CSV column header name in DUNS file
+column_headers = [
+    "awardee_or_recipient_uniqu",  # DUNS Field
+    "registration_date",  # Registration_Date
+    "expiration_date",  # Expiration_Date
+    "last_sam_mod_date",  # Last_Update_Date
+    "activation_date",  # Activation_Date
+    "legal_business_name"  # Legal_Business_Name
+]
+
+
+def remove_existing_duns(data, sess):
+    """Remove rows frome file that alread have a entry in broker database. We should only update missing DUNS"""
+
+    duns_in_file = ",".join(list(data['awardee_or_recipient_uniqu'].unique()))
+    sql_query = "SELECT awardee_or_recipient_uniqu " +\
+                "FROM duns where awardee_or_recipient_uniqu = ANY('{" + \
+                duns_in_file +\
+                "}')"
+
+    db_duns = pd.read_sql(sql_query, sess.bind)
+    missing_duns = data[~data['awardee_or_recipient_uniqu'].isin(db_duns['awardee_or_recipient_uniqu'])]
+
+    return missing_duns
+
+
+def clean_duns_csv_data(data):
+    return clean_data(data, DUNS, {
+        x: x for x in column_headers
+    }, {})
+
+
+def run_duns_batches(file, sess, block_size=10000, batch=0):
+    """Updates DUNS table in batches from csv file"""
+    logger.info("Retrieving total rows from duns file")
+    start = datetime.now()
+    row_count = len(pd.read_csv(file, sep='",\s"', engine='python'))
```

Comment:
Should this have the same separator as the call below? Or visa versa?

Reply:
Yes it should for consistency.  Updating in next commit.